pkg/providers/provider.go

```diff
@@ -14,6 +14,7 @@ type Provider interface {
 	SetupAndValidateCreateCluster(ctx context.Context, clusterSpec *cluster.Spec) error
 	SetupAndValidateDeleteCluster(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error
 	SetupAndValidateUpgradeCluster(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec, currentSpec *cluster.Spec) error
+	SetupUpgradeManagementComponents(ctx context.Context, clusterSpec *cluster.Spec) error
```

Comment:
Why is this "setup"?
If the goal is to validate before `EnvMap` is called by clusterctl, why not just call this `ValidateManagementComponentsConfig`

Reply:
We also sets environment variables, so I consider this to be a "setup" as well
E.g. `VSPHERE_USERNAME` is set if `EKSA_VSPHERE_USERNAME` exists.

Reply:
Why do we do that? Those variables are already passed to clusterctl
Setting env vars like that is not very great
If they are needed, it represents a hidden dependency

Reply:
I'm not sure why we do that exactly. It seems as though we have a set of environment variables that we expose to the user for input, but another set that we use when passing them to clusterctl.

We could perhaps change this behavior to have `providers.EnvMap` read the input variables directly instead if it's a blocker.

Reply:
Address through slack. We'll create a ticket to address this and come back to it.

A link to the issue has been added under one of the affected providers (vsphere).